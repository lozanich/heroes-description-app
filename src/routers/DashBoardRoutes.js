import React from "react";
import { Switch, Route, Redirect } from "react-router-dom";
import { DcScreen } from "../components/dc/DcScreen";
import { HeroScreen } from "../components/heroes/HeroScreen";
import { MarvelScreen } from "../components/marvel/MarvelScreen";
import { SearchScreen } from "../components/search/SearchScreen";
import { Navbar } from "../components/ui/Navbar";

export const DashBoardRoutes = () => {
  return (
    <>
      <Navbar />

      <div className="container">
        <Switch>
          <Route exact path="/marvel" component={MarvelScreen} />
          <Route exact path="/hero/:heroeId" component={HeroScreen} />
          <Route exact path="/dc" component={DcScreen} />
          <Route exact path="/searchhero" component={SearchScreen} />

          <Redirect to="/marvel" />
        </Switch>
      </div>
    </>
  );
};
