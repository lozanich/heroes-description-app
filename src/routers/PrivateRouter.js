import React from "react";
import PropTypes from "prop-types";
import { Redirect, Route } from "react-router-dom";

export const PrivateRouter = ({ isAuth, component: Component, ...rest }) => {
  //   console.log(rest.location.pathname);

  localStorage.setItem("lastPath", rest.location.pathname);

  return (
    <Route
      {...rest}
      component={(props) =>
        isAuth ? <Component {...props} /> : <Redirect to="/login" />
      }
    />
  );
};

PrivateRouter.propTypes = {
  isAuth: PropTypes.bool.isRequired,
  component: PropTypes.func.isRequired,
};
