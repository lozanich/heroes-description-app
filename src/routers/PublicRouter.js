import React from "react";
import { Redirect, Route } from "react-router-dom";
import PropTypes from "prop-types";

export const PublicRouter = ({ isAuth, component: Component, ...rest }) => {
  // rest son el resto de los elementos o propiedades que se pasan como argumento
  // atravez de las props

  return (
    <Route
      {...rest}
      component={(props) =>
        isAuth ? <Redirect to="/" /> : <Component {...props} />
      }
    />
  );
};

PublicRouter.propTypes = {
  isAuth: PropTypes.bool.isRequired,
  component: PropTypes.func.isRequired,
};
