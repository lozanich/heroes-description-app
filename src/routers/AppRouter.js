import React, { useContext } from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import { AuthContext } from "../auth/AuthContext";
import { LoginScreen } from "../components/login/LoginScreen";
import { DashBoardRoutes } from "./DashBoardRoutes";
import { PrivateRouter } from "./PrivateRouter";
import { PublicRouter } from "./PublicRouter";

export const AppRouter = () => {
  const { user } = useContext(AuthContext);
  const { logged } = user;

  // console.log(user);

  return (
    <Router>
      <div>
        {/* <Navbar /> */}

        {/* A <Switch> looks through its children <Route>s and
            renders the first one that matches the current URL. */}
        <Switch>
          {/* <Route path="/marvel">
            <Marvel />
          </Route>
          <Route path="/dc">
            <DcScreen />
          </Route> */}
          <PublicRouter
            isAuth={logged}
            component={LoginScreen}
            exact
            path="/login"
          />

          <PrivateRouter isAuth={logged} path="/" component={DashBoardRoutes} />
        </Switch>
      </div>
    </Router>
  );
};
