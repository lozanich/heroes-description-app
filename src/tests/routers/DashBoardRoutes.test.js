import React from "react";
import "@testing-library/jest-dom";
import { mount, shallow } from "enzyme";
import { DashBoardRoutes } from "../../routers/DashBoardRoutes";
import { AuthContext } from "../../auth/AuthContext";
import { MemoryRouter } from "react-router-dom";

describe("Pruebas en el componente <DashBoardRoutes />", () => {
  const contextValue = {
    dispatch: jest.fn(),
    user: {
      logged: true,
    },
  };

  test("Debe de mostrar correctamente", () => {
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <MemoryRouter>
          <DashBoardRoutes />
        </MemoryRouter>
      </AuthContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
