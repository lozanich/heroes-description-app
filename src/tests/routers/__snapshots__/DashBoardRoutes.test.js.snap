// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <DashBoardRoutes /> Debe de mostrar correctamente 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/marvel"
          onClick={[Function]}
          style={Object {}}
        >
          Marvel
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/searchhero"
          onClick={[Function]}
        >
          Buscar heroe
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link text-info"
        />
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container"
  >
    Array [
      <h1>
        Marvel
      </h1>,
      <hr />,
      <div
        className="card-columns"
      >
        <div
          className="card animate__animated animate__jello"
          style={
            Object {
              "maxWidth": 500,
              "minHeight": 840,
              "width": "18rem",
            }
          }
        >
          <img
            alt="Spider Man"
            className="card-img-top"
            src="./assets/heroes/marvel-spider.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Spider Man
            </h5>
            <ul
              className="list-group list-group-flush"
            >
              <li
                className="list-group-item"
              >
                Marvel Comics
              </li>
              <li
                className="list-group-item"
              >
                Amazing Fantasy #15
              </li>
            </ul>
            <a
              className="btn btn-primary"
              href="/hero/marvel-spider"
              onClick={[Function]}
            >
              Detalles
            </a>
          </div>
        </div>
        <div
          className="card animate__animated animate__jello"
          style={
            Object {
              "maxWidth": 500,
              "minHeight": 840,
              "width": "18rem",
            }
          }
        >
          <img
            alt="Captain America"
            className="card-img-top"
            src="./assets/heroes/marvel-captain.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Captain America
            </h5>
            <ul
              className="list-group list-group-flush"
            >
              <li
                className="list-group-item"
              >
                Marvel Comics
              </li>
              <li
                className="list-group-item"
              >
                Captain America Comics #1
              </li>
            </ul>
            <a
              className="btn btn-primary"
              href="/hero/marvel-captain"
              onClick={[Function]}
            >
              Detalles
            </a>
          </div>
        </div>
        <div
          className="card animate__animated animate__jello"
          style={
            Object {
              "maxWidth": 500,
              "minHeight": 840,
              "width": "18rem",
            }
          }
        >
          <img
            alt="Iron Man"
            className="card-img-top"
            src="./assets/heroes/marvel-iron.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Iron Man
            </h5>
            <ul
              className="list-group list-group-flush"
            >
              <li
                className="list-group-item"
              >
                Marvel Comics
              </li>
              <li
                className="list-group-item"
              >
                Tales of Suspense #39
              </li>
            </ul>
            <a
              className="btn btn-primary"
              href="/hero/marvel-iron"
              onClick={[Function]}
            >
              Detalles
            </a>
          </div>
        </div>
        <div
          className="card animate__animated animate__jello"
          style={
            Object {
              "maxWidth": 500,
              "minHeight": 840,
              "width": "18rem",
            }
          }
        >
          <img
            alt="Thor"
            className="card-img-top"
            src="./assets/heroes/marvel-thor.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Thor
            </h5>
            <ul
              className="list-group list-group-flush"
            >
              <li
                className="list-group-item"
              >
                Marvel Comics
              </li>
              <li
                className="list-group-item"
              >
                Journey into Myster #83
              </li>
            </ul>
            <a
              className="btn btn-primary"
              href="/hero/marvel-thor"
              onClick={[Function]}
            >
              Detalles
            </a>
          </div>
        </div>
        <div
          className="card animate__animated animate__jello"
          style={
            Object {
              "maxWidth": 500,
              "minHeight": 840,
              "width": "18rem",
            }
          }
        >
          <img
            alt="Hulk"
            className="card-img-top"
            src="./assets/heroes/marvel-hulk.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Hulk
            </h5>
            <ul
              className="list-group list-group-flush"
            >
              <li
                className="list-group-item"
              >
                Marvel Comics
              </li>
              <li
                className="list-group-item"
              >
                The Incredible Hulk #1
              </li>
            </ul>
            <a
              className="btn btn-primary"
              href="/hero/marvel-hulk"
              onClick={[Function]}
            >
              Detalles
            </a>
          </div>
        </div>
        <div
          className="card animate__animated animate__jello"
          style={
            Object {
              "maxWidth": 500,
              "minHeight": 840,
              "width": "18rem",
            }
          }
        >
          <img
            alt="Wolverine"
            className="card-img-top"
            src="./assets/heroes/marvel-wolverine.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Wolverine
            </h5>
            <ul
              className="list-group list-group-flush"
            >
              <li
                className="list-group-item"
              >
                Marvel Comics
              </li>
              <li
                className="list-group-item"
              >
                The Incredible Hulk #180
              </li>
            </ul>
            <a
              className="btn btn-primary"
              href="/hero/marvel-wolverine"
              onClick={[Function]}
            >
              Detalles
            </a>
          </div>
        </div>
        <div
          className="card animate__animated animate__jello"
          style={
            Object {
              "maxWidth": 500,
              "minHeight": 840,
              "width": "18rem",
            }
          }
        >
          <img
            alt="Daredevil"
            className="card-img-top"
            src="./assets/heroes/marvel-daredevil.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Daredevil
            </h5>
            <ul
              className="list-group list-group-flush"
            >
              <li
                className="list-group-item"
              >
                Marvel Comics
              </li>
              <li
                className="list-group-item"
              >
                Daredevil #1
              </li>
            </ul>
            <a
              className="btn btn-primary"
              href="/hero/marvel-daredevil"
              onClick={[Function]}
            >
              Detalles
            </a>
          </div>
        </div>
        <div
          className="card animate__animated animate__jello"
          style={
            Object {
              "maxWidth": 500,
              "minHeight": 840,
              "width": "18rem",
            }
          }
        >
          <img
            alt="Hawkeye"
            className="card-img-top"
            src="./assets/heroes/marvel-hawkeye.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Hawkeye
            </h5>
            <ul
              className="list-group list-group-flush"
            >
              <li
                className="list-group-item"
              >
                Marvel Comics
              </li>
              <li
                className="list-group-item"
              >
                Tales of Suspense #57
              </li>
            </ul>
            <a
              className="btn btn-primary"
              href="/hero/marvel-hawkeye"
              onClick={[Function]}
            >
              Detalles
            </a>
          </div>
        </div>
        <div
          className="card animate__animated animate__jello"
          style={
            Object {
              "maxWidth": 500,
              "minHeight": 840,
              "width": "18rem",
            }
          }
        >
          <img
            alt="Cyclops"
            className="card-img-top"
            src="./assets/heroes/marvel-cyclops.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Cyclops
            </h5>
            <ul
              className="list-group list-group-flush"
            >
              <li
                className="list-group-item"
              >
                Marvel Comics
              </li>
              <li
                className="list-group-item"
              >
                X-Men #1
              </li>
            </ul>
            <a
              className="btn btn-primary"
              href="/hero/marvel-cyclops"
              onClick={[Function]}
            >
              Detalles
            </a>
          </div>
        </div>
        <div
          className="card animate__animated animate__jello"
          style={
            Object {
              "maxWidth": 500,
              "minHeight": 840,
              "width": "18rem",
            }
          }
        >
          <img
            alt="Silver Surfer"
            className="card-img-top"
            src="./assets/heroes/marvel-silver.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Silver Surfer
            </h5>
            <ul
              className="list-group list-group-flush"
            >
              <li
                className="list-group-item"
              >
                Marvel Comics
              </li>
              <li
                className="list-group-item"
              >
                The Fantastic Four #48
              </li>
            </ul>
            <a
              className="btn btn-primary"
              href="/hero/marvel-silver"
              onClick={[Function]}
            >
              Detalles
            </a>
          </div>
        </div>
      </div>,
    ]
  </div>,
]
`;
